src/main/java/com/aws/greengrass/util/FileSystemPermission.java

```diff
@@ -70,6 +82,90 @@
         return ret;
     }
 
+    /**
+     * Convert to a list of Acl entries for use with AclFileAttributeView.setAcl.
+     *
+     * @param path path to apply to
+     * @return List of Acl entries
+     * @throws IOException if any exception occurs while converting to Acl
+     */
+    public List<AclEntry> toAclEntries(Path path) throws IOException {
+        // These sets of permissions are reverse engineered by setting the permission on a file using Windows File
+        // Explorer. Then use AclFileAttributeView.getAcl to examine what were set.
+        final Set<AclEntryPermission> readPerms = new HashSet<>(Arrays.asList(
+                AclEntryPermission.READ_DATA,
+                AclEntryPermission.READ_NAMED_ATTRS,
+                AclEntryPermission.READ_ATTRIBUTES,
+                AclEntryPermission.READ_ACL,
+                AclEntryPermission.SYNCHRONIZE));
+        final Set<AclEntryPermission> writePerms = new HashSet<>(Arrays.asList(
+                AclEntryPermission.WRITE_DATA,
+                AclEntryPermission.APPEND_DATA,
+                AclEntryPermission.WRITE_NAMED_ATTRS,
+                AclEntryPermission.WRITE_ATTRIBUTES,
+                AclEntryPermission.SYNCHRONIZE));
+        final Set<AclEntryPermission> executePerms = new HashSet<>(Arrays.asList(
+                AclEntryPermission.READ_DATA,
+                AclEntryPermission.READ_NAMED_ATTRS,
+                AclEntryPermission.EXECUTE,
+                AclEntryPermission.READ_ATTRIBUTES,
+                AclEntryPermission.READ_ACL,
+                AclEntryPermission.SYNCHRONIZE));
+
+        UserPrincipalLookupService userPrincipalLookupService = path.getFileSystem().getUserPrincipalLookupService();
+        UserPrincipal ownerPrincipal = userPrincipalLookupService.lookupPrincipalByName(ownerUser);
+        GroupPrincipal everyone = userPrincipalLookupService.lookupPrincipalByGroupName("Everyone");
+
+        List<AclEntry> aclEntries = new ArrayList<>();
+        if (ownerRead) {
+            aclEntries.add(AclEntry.newBuilder()
+                    .setType(AclEntryType.ALLOW)
+                    .setPrincipal(ownerPrincipal)
+                    .setPermissions(readPerms)
+                    .build());
+        }
+        if (ownerWrite) {
+            aclEntries.add(AclEntry.newBuilder()
+                    .setType(AclEntryType.ALLOW)
+                    .setPrincipal(ownerPrincipal)
+                    .setPermissions(writePerms)
+                    .build());
+        }
+        if (ownerExecute) {
+            aclEntries.add(AclEntry.newBuilder()
+                    .setType(AclEntryType.ALLOW)
+                    .setPrincipal(ownerPrincipal)
+                    .setPermissions(executePerms)
+                    .build());
+        }
+
+        // There is no default group concept on Windows. (There is, but is used when mounting as a network share.)
```

Comment:
`groupRead`, `groupWrite`, `groupExecute` are not used. Windows does not natively has a default group concept. Although administrator can set a default group WHEN a filesystem is being used as a network share that appears as POSIX, but this does not apply here.